numpy/_core/src/common/simd/lsx/math.h

```diff
@@ -0,0 +1,228 @@
+#define _NPY_SIMD_LSX_MATH_H
+/***************************
+ * Elementary
+ ***************************/
+// Square root
+#define npyv_sqrt_f32 __lsx_vfsqrt_s
+#define npyv_sqrt_f64 __lsx_vfsqrt_d
+
+// Reciprocal
+NPY_FINLINE npyv_f32 npyv_recip_f32(npyv_f32 a)
+{ return __lsx_vfrecip_s(a); }
+NPY_FINLINE npyv_f64 npyv_recip_f64(npyv_f64 a)
+{ return __lsx_vfrecip_d(a); }
+
+// Absolute
+NPY_FINLINE npyv_f32 npyv_abs_f32(npyv_f32 a)
+{
+  return (npyv_f32)__lsx_vbitclri_w(a, 0x1F);
+}
+NPY_FINLINE npyv_f64 npyv_abs_f64(npyv_f64 a)
+{
+  return (npyv_f64)__lsx_vbitclri_d(a, 0x3F);
+}
+
+// Square
+NPY_FINLINE npyv_f32 npyv_square_f32(npyv_f32 a)
+{ return __lsx_vfmul_s(a, a); }
+NPY_FINLINE npyv_f64 npyv_square_f64(npyv_f64 a)
+{ return __lsx_vfmul_d(a, a); }
+
+// Maximum, natively mapping with no guarantees to handle NaN.
+#define npyv_max_f32 __lsx_vfmax_s
+#define npyv_max_f64 __lsx_vfmax_d
+// Maximum, supports IEEE floating-point arithmetic (IEC 60559),
+// - If one of the two vectors contains NaN, the equivalent element of the other vector is set
+// - Only if both corresponded elements are NaN, NaN is set.
+NPY_FINLINE npyv_f32 npyv_maxp_f32(npyv_f32 a, npyv_f32 b)
+{
+  return __lsx_vfmax_s(a, b);
+}
+NPY_FINLINE npyv_f64 npyv_maxp_f64(npyv_f64 a, npyv_f64 b)
+{
+  return __lsx_vfmax_d(a, b);
+}
+// If any of corresponded element is NaN, NaN is set.
+NPY_FINLINE npyv_f32 npyv_maxn_f32(npyv_f32 a, npyv_f32 b)
+{
+  __m128i mask = __lsx_vand_v(npyv_notnan_f32(a), npyv_notnan_f32(b));
+  __m128 max   = __lsx_vfmax_s(a, b);
+  return npyv_select_f32(mask, max, (__m128){NAN, NAN, NAN, NAN});
+}
+NPY_FINLINE npyv_f64 npyv_maxn_f64(npyv_f64 a, npyv_f64 b)
+{
+  __m128i mask = __lsx_vand_v(npyv_notnan_f64(a), npyv_notnan_f64(b));
+  __m128d max  = __lsx_vfmax_d(a, b);
+  return npyv_select_f64(mask, max, (__m128d){NAN, NAN});
+}
+
+// Maximum, integer operations
+#define npyv_max_u8  __lsx_vmax_bu
+#define npyv_max_s8  __lsx_vmax_b
+#define npyv_max_u16 __lsx_vmax_hu
+#define npyv_max_s16 __lsx_vmax_h
+#define npyv_max_u32 __lsx_vmax_wu
+#define npyv_max_s32 __lsx_vmax_w
+#define npyv_max_u64 __lsx_vmax_du
+#define npyv_max_s64 __lsx_vmax_d
+
+// Minimum, natively mapping with no guarantees to handle NaN.
+#define npyv_min_f32 __lsx_vfmin_s
+#define npyv_min_f64 __lsx_vfmin_d
+
+// Minimum, supports IEEE floating-point arithmetic (IEC 60559),
+// - If one of the two vectors contains NaN, the equivalent element of the other vector is set
+// - Only if both corresponded elements are NaN, NaN is set.
+NPY_FINLINE npyv_f32 npyv_minp_f32(npyv_f32 a, npyv_f32 b)
+{
+  return __lsx_vfmin_s(a, b);
+}
+NPY_FINLINE npyv_f64 npyv_minp_f64(npyv_f64 a, npyv_f64 b)
+{
+  return __lsx_vfmin_d(a, b);
+}
+NPY_FINLINE npyv_f32 npyv_minn_f32(npyv_f32 a, npyv_f32 b)
+{
+  __m128i mask = __lsx_vand_v(npyv_notnan_f32(a), npyv_notnan_f32(b));
+  __m128 min   = __lsx_vfmin_s(a, b);
+  return npyv_select_f32(mask, min, (__m128){NAN, NAN, NAN, NAN});
```

Comment:
```suggestion
  return __lsx_vfmina_s(a, b);
```
`__lsx_vfmina_s` should propagates NaNs, isn't?